### ISSUE TITLE
Cannot remove files used by GWT project

### ISSUE THREAD

**Issue by GryfonN** [gwt-plugins/gwt-eclipse-plugin#325]
After I migrated from old _Google Plugin for Eclipse_ to new _GWT Eclipse Plugin_ I have a problem to delete JAR file that is used by my GWT project in EclipseIDE. When I try to remove this file, I have "File In Use" warning dialog in windows7 that says me "The action can't be completed because the file is open in eclipse.exe. Close the file and try again." I also can not delete this JAR file in EclipseIDE.

This JAR library is removed and one more time created by some external process that collects some resources from my disk and ZIP them together for my GWT project.
The references for this JAR are:
- in **.classpath** file as classpathentry, because it is a dependency in _Project Properties > Java Build Path > Tab: Libraries_ as JAR from another project
- in **.settings/com.gwtplugins.gdt.eclipse.core.prefs** file as classpathentry, because it is build path entry in _Project Properties > GWT > Web Application_

With _Google Plugin for Eclipse_ there was no problem to delete this JAR file, after that I had of course red problems in EclipseIDE, but when new JAR was created on my disk, I refreshed the GWT project and everything was fine again.

**eclipse-SDK-4.6.1
GWT Eclipse Plugin 	3.0.0.201611042254**

**Comment by branflake2267** [comment]
Hm, sounds like there was a latent file lock on the jar that couldn't be deleted. This I think is probably unrelated to the GWT Eclipse plugin and more related to some process in Eclipse IDE or Windows OS. Maybe a validation process was reading and locked the file. Not sure what to do about this yet.

**Comment by GryfonN** [comment]
I have two EclipseIDEs with same plugins/preferences on same Win7, only difference is new GWT Eclipse Plugin and old Google Plugin for Eclipse (and .classpath / .settings files for these plugins)

**Comment by GryfonN** [comment]
Ok I have found the real reason for my problem. **I am sorry, it was nothing with GWT Eclipse Plugin at all**, so I will close this issue and describe my problem little bit more.

The source of my problem was not correctly closed stream from used JAR, and because of that I was not able to delete it, even when Eclipse GWT project was not running anymore. Reason for that was **Xerces** library in gwt-dev.jar. I have no idea why I had not this problem with GWT 2.7 (maybe different position of <classpathentry> in .classpath file of Eclipse project could affect it), but when I upgraded it to GWT 2.8 (also with migration from Google GWT plugin to this one) I had
_ClassCastException: org.apache.xerces.parsers.XIncludeAwareParserConfiguration cannot be cast to org.apache.xerces.xni.parser.XMLParserConfiguration_ in my log and this pointed me the right problem.

The reason for that exception was that before with such code like: 
```java
DocumentBuilderFactory newInstance = DocumentBuilderFactory.newInstance();
DocumentBuilder newDocumentBuilder = newInstance.newDocumentBuilder();
baseLayoutXmlDocument = newDocumentBuilder.parse( baseLayoutSvgInputStream );

SAXParserFactory spf = SAXParserFactory.newInstance();
SAXParser sp = spf.newSAXParser();
```

the implementations from JDK package _com.sun.org.apache.xerces.internal.jaxp_ was used, but after upgrade to GWT2.8 my app chose the xerces from gwt-dev.jar, java code raised the exception, did not closed the streams correctly, and that was the reason why eclipse.exe process was still using my JAR from disk.

Just FYI I found easy fix for that for now https://www.databasesandlife.com/java-always-explicitly-specify-which-xml-parser-to-use/
```bash
-Djavax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl
-Djavax.xml.parsers.SAXParserFactory=com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl
```
